static/js/lib/queries/applications.js

```diff
@@ -126,5 +128,18 @@ export default {
         "X-CSRFTOKEN": getCookie("csrftoken")
       }
     }
+  }),
+  applicationUploadResumeQuery: (
+    applicationDetail: ApplicationResumeDetail,
+    applicationId: number
+  ) => ({
+    queryKey: "resume",
+    url:      `/api/applications/${applicationId}/resume/`,
+    body:     {
+      ...applicationDetail
+    },
+    options: {
```

Comment:
If we add a return value to the serializer, we can configure this query in such a way that the resume filename will show up immediately in the drawer after it successfully uploads. We'll need to do 2 things:

1. In `applications.views.UploadResumeView.post`, return all of the resume-relevant properties in the response body (`resume_filepath`, `linkedin_url`, `resume_upload_date`). For `resume_filepath` you can move that logic out of the serializer and just add it as a property on the `Application` model itself. We may also want a python test to make sure that those property names are the same in the response from this endpoint and from the application detail endpoint.
1. Add `transform` and `update` properties on this query config. Basically we would just need to update the `applicationDetail` entity state. That state has application ids mapped to an `ApplicationDetail`-type object. I _think_ something like this would work:
    ```javascript
    transform: (json: ResumeUploadResponse) => {
      return {
        [applicationDetailKey]: {
          // The json is not a full `ApplicationDetail` object. Just has some
          // of the keys. That's hopefully ok since we're using mergeDeepRight below in update()
          [applicationId]: json
        }
      }
    },
    update: {
      [applicationDetailKey]: (
        prev: ApplicationDetailState,
        transformed: Object // we could create a type for this too if needed
      ) => (
        R.mergeDeepRight(
          prev,
          transformed
        ) 
      )
    }
    ```
After that, `state.entities.applicationDetail` should look the same, except the resume properties for the logged-in user's detail should be updated, and the uploaded resume file path should be viewable in the drawer right away.